undefined reference to `ucm_set_global_opts'

I am seeing the following error when building UCC in NVIDIA's PyTorch container:
```
  CCLD     ucc_info
/usr/bin/ld: /usr/local/lib/libucs.so: undefined reference to `ucm_set_global_opts'
/usr/bin/ld: /usr/local/lib/libucs.so: undefined reference to `ucm_mmap_hook_modes'
```

To reproduce, run the following docker container:
```
docker run -it nvcr.io/nvidia/pytorch:22.03-py3
```
And inside the container, run the following script:
```bash
#!/bin/bash

set -ex

export UCX_HOME="/usr"
export UCC_HOME="/usr"

install_ucx() {
    set -ex
    echo "Will install ucx at: $UCX_HOME"
    rm -rf ucx
    git clone --recursive https://github.com/openucx/ucx.git
    pushd ucx
    ./autogen.sh
    ./configure --prefix=$UCX_HOME      \
        --without-bfd                   \
        --enable-mt                     \
        --with-cuda=/usr/local/cuda/    \
        --enable-profiling              \
        --enable-stats
    make -j
    make install
    popd
}

install_ucc() {
    set -ex
    echo "Will install ucc at: $UCC_HOME"
    rm -rf ucc
    git clone --recursive https://github.com/openucx/ucc.git
    pushd ucc
    ./autogen.sh
    ./configure --prefix=$UCC_HOME      \
        --with-ucx=$UCX_HOME            \
        --with-nccl=/usr                \
        --with-cuda=/usr/local/cuda/
    make -j
    make install
    popd
}

install_ucx
install_ucc
```

--- a/config/m4/ucx.m4
+++ b/config/m4/ucx.m4
@@ -56,7 +56,7 @@ AS_IF([test "x$ucx_checked" != "xyes"],[
         [
             ucx_happy="no"
         ])
-        
+
         AC_CHECK_HEADERS([ucs/sys/uid.h],
         [
             AC_CHECK_LIB([ucs], [ucs_get_system_id],
@@ -88,6 +88,7 @@ AS_IF([test "x$ucx_checked" != "xyes"],[
             [
                 AC_SUBST(UCX_LDFLAGS, "-L$check_ucx_libdir")
                 AC_SUBST(UCS_LDFLAGS, "-L$check_ucx_libdir")
+                AC_SUBST(UCS_LIBDIR, $check_ucx_libdir)
             ])
 
             AC_SUBST(UCX_LIBADD, "-lucp -lucm")

--- a/test/gtest/Makefile.am
+++ b/test/gtest/Makefile.am
@@ -55,7 +55,9 @@ gtest_CPPFLAGS = \
 	-I$(top_srcdir)/test/gtest \
 	$(GTEST_CPPFLAGS)
 
-gtest_LDFLAGS  = $(GTEST_LDFLAGS) -pthread -no-install -Wl,-dynamic-list-data
+gtest_LDFLAGS = $(GTEST_LDFLAGS) -pthread -no-install -Wl,-dynamic-list-data \
+    -Wl,--rpath-link=${UCS_LIBDIR}
+
 gtest_CFLAGS   = $(BASE_CFLAGS) $(AM_CPPFLAGS)
 gtest_CXXFLAGS = -std=gnu++11 \
 	$(BASE_CXXFLAGS) $(GTEST_CXXFLAGS) \

--- a/test/mpi/Makefile.am
+++ b/test/mpi/Makefile.am
@@ -32,12 +32,13 @@ ucc_test_mpi_SOURCES = \
 
 CXX=$(MPICXX)
 LD=$(MPICXX)
-ucc_test_mpi_CPPFLAGS=$(BASE_CPPFLAGS)
-ucc_test_mpi_CXXFLAGS=$(BASE_CXXFLAGS) -std=gnu++11
-ucc_test_mpi_LDADD=$(UCC_TOP_BUILDDIR)/src/libucc.la
+ucc_test_mpi_CPPFLAGS = $(BASE_CPPFLAGS)
+ucc_test_mpi_CXXFLAGS = $(BASE_CXXFLAGS) -std=gnu++11
+ucc_test_mpi_LDFLAGS = -Wl,--rpath-link=${UCS_LIBDIR}
+ucc_test_mpi_LDADD = $(UCC_TOP_BUILDDIR)/src/libucc.la
 
 if HAVE_CUDA
-ucc_test_mpi_CPPFLAGS+=$(CUDA_CPPFLAGS)
-ucc_test_mpi_LDFLAGS=$(CUDA_LDFLAGS)
-ucc_test_mpi_LDADD+=$(CUDA_LIBS)
+ucc_test_mpi_CPPFLAGS += $(CUDA_CPPFLAGS)
+ucc_test_mpi_LDFLAGS += $(CUDA_LDFLAGS)
+ucc_test_mpi_LDADD += $(CUDA_LIBS)
 endif

--- a/tools/info/Makefile.am
+++ b/tools/info/Makefile.am
@@ -28,5 +28,6 @@ nodist_ucc_info_SOURCES = \
 
 ucc_info_CPPFLAGS = $(AM_CPPFLAGS) $(BASE_CPPFLAGS) -I${UCC_TOP_BUILDDIR}/src
 ucc_info_CFLAGS = $(BASE_CFLAGS)
+ucc_info_LDFLAGS = -Wl,--rpath-link=${UCS_LIBDIR}
 ucc_info_LDADD = \
 	$(UCC_TOP_BUILDDIR)/src/libucc.la

--- a/tools/perf/Makefile.am
+++ b/tools/perf/Makefile.am
@@ -28,12 +28,13 @@ ucc_perftest_SOURCES =            \
 
 CXX=$(MPICXX)
 LD=$(MPICXX)
-ucc_perftest_CPPFLAGS=$(BASE_CPPFLAGS)
-ucc_perftest_CXXFLAGS=-std=gnu++11 $(BASE_CXXFLAGS)
-ucc_perftest_LDADD=$(UCC_TOP_BUILDDIR)/src/libucc.la
+ucc_perftest_CPPFLAGS = $(BASE_CPPFLAGS)
+ucc_perftest_CXXFLAGS = -std=gnu++11 $(BASE_CXXFLAGS)
+ucc_perftest_LDFLAGS = -Wl,--rpath-link=${UCS_LIBDIR}
+ucc_perftest_LDADD = $(UCC_TOP_BUILDDIR)/src/libucc.la
 
 if HAVE_CUDA
-ucc_perftest_CPPFLAGS+=$(CUDA_CPPFLAGS)
-ucc_perftest_LDFLAGS=$(CUDA_LDFLAGS)
-ucc_perftest_LDADD+=$(CUDA_LIBS)
+ucc_perftest_CPPFLAGS += $(CUDA_CPPFLAGS)
+ucc_perftest_LDFLAGS += $(CUDA_LDFLAGS)
+ucc_perftest_LDADD += $(CUDA_LIBS)
 endif

